moss.go

```diff
@@ -0,0 +1,115 @@
+package gomoss
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+)
+
+// MossRequest is structure
+type MossRequest struct {
+	// ID is the Moss server user ID
+	ID uint32
+	// Language is programing language type
+	Language string
+	// Ignore limit
+	IgnoreLimit uint16
+	// Comment is comment that will appear on result
+	Comment string
+	// ExperimentalServer is flag utilize experimental server
+	ExperimentalServer uint8
+	// MatchingNumber is number of matching files to show in the result
+	MatchingNumber uint16
+	// DirectoryMode is a mode to add files with directory
+	DirectoryMode uint8
+	// Code is source codes that will be sent to Moss server
+	Code code
+	// BaseCode is base codes that will be sent to Moss server
+	BaseCode code
+}
+
+// code struct
+type code struct {
+	// reader is opened file reader
+	reader []io.Reader
```

Comment:
can this be a single `io.Reader` and a single `string`?

```go
type code struct {
  reader io.Reader
  displayName string
}

type codeList struct {
    list []*code
}

type MossRequest struct {
    ...
    Code codeList
    BaseCode codeList
}

// Add adds Code into the request.
func (c *codeList) Add(reader io.Reader, displayName string) {
	c.list = append(c.list, &code{reader, displayName})
}
```

that way `uploadFile` can take a `code`.